src/core/api.ts

```diff
@@ -1630,19 +1630,19 @@ export class Api<R extends RawApi = RawApi> {
     }
 
     /**
-     * Use this method to set the thumbnail of a sticker set. Animated thumbnails can be set for animated sticker sets only. Returns True on success.
+     * Use this method to set the thumbnail of a sticker set. Animated thumbnails can be set for animated sticker sets only. Video thumbnails can be set only for video sticker sets only. Returns True on success.
      *
      * @param name Sticker set name
      * @param user_id User identifier of the sticker set owner
-     * @param thumb A PNG image with the thumbnail, must be up to 128 kilobytes in size and have width and height exactly 100px, or a TGS animation with the thumbnail up to 32 kilobytes in size; see https://core.telegram.org/animated_stickers#technical-requirements for animated sticker technical requirements. Pass a file_id as a String to send a file that already exists on the Telegram servers, pass an HTTP URL as a String for Telegram to get a file from the Internet, or upload a new one using multipart/form-data.. Animated sticker set thumbnail can't be uploaded via HTTP URL.
+     * @param thumb A PNG image with the thumbnail, must be up to 128 kilobytes in size and have width and height exactly 100px, or a TGS animation with the thumbnail up to 32 kilobytes in size; see https://core.telegram.org/stickers#animated-sticker-requirements for animated sticker technical requirements, or a WEBM video with the thumbnail up to 32 kilobytes in size; see https://core.telegram.org/stickers#video-sticker-requirements for video sticker technical requirements. Pass a file_id as a String to send a file that already exists on the Telegram servers, pass an HTTP URL as a String for Telegram to get a file from the Internet, or upload a new one using multipart/form-data. More info on Sending Files ». Animated sticker set thumbnails can't be uploaded via HTTP URL.
      * @param signal Optional `AbortSignal` to cancel the request
      *
      * **Official reference:** https://core.telegram.org/bots/api#setstickersetthumb
      */
     setStickerSetThumb(
         name: string,
         user_id: number,
-        thumb: InputFile | string,
+        thumb?: InputFile | string,
```

Comment:
oh, maybe I have overlooked something.

setThumb which has an optional thumb? Method feels a bit strange this way?

Reply:
Its the same in the official docs but I don't understand why that is. The text doesn't describe why it should be optional.

If it is able to remove a thumbnail that should be more explicit to not forget the argument:

```suggestion
        thumb: InputFile | string | undefined,
```

Reply:
Yeah I thought about this. On the other hand, `Api` is about making it convenient. I would find it odd to have to pass `{ thumb: new InputFile() }` at this point.

There are other methods with only a single optional argument which do the same thing. It just does not make much sense.

The only real case where this can look awkward is when removing the thumbnail but wanting to pass a signal to abort the request, in which case it becomes `setStickerSetThumb(name, id, undefined, signal)`. I still find that readable for what it does.

Reply:
Especially considering that the alternative would be to pass `{}` instead of `undefined`.

Can I release?

Reply:
Specifying `undefined` over omitting it makes it clear that this is the wanted behaviour and not something that was forgotten to set.

Guess I am fine either way, but it should be consistent. This should be changed then: https://github.com/grammyjs/grammY/blob/d5731cbe53d9a5ad753656bbcb9226bb8283b956/src/core/api.ts#L1056

Reply:
> Its the same in the official docs but I don't understand why that is. The text doesn't describe why it should be optional.
> 
> If it is able to remove a thumbnail that should be more explicit to not forget the argument:

I find it weird to interdict something at compile-time that works at runtime. It is perfectly okay not to pass the argument.

This is also the case if we were to introduce an options object, which can always be omitted.

Reply:
> This should be changed then:
> 
> https://github.com/grammyjs/grammY/blob/d5731cbe53d9a5ad753656bbcb9226bb8283b956/src/core/api.ts#L1056

Good point! I'll fix it and then release.